packages/@react-stately/collections/src/getItemCount.ts

```diff
@@ -25,11 +25,16 @@ export function getItemCount<T>(collection: Collection<Node<T>>): number {
   let counter = 0;
   let countItems = (items: Iterable<Node<T>>) => {
     for (let item of items) {
-      if (item.type === 'section') {
-        countItems(getChildNodes(item, collection));
-      } else if (item.type === 'item') {
+      if (isValidItem(item)) {
         counter++;
       }
+
+      // TODO: New collections vs old collection is different here. New collections for table will only return
```

Comment:
still a TODO here? or are things working and you just want alternatives?

Reply:
The latter, feels ok for now but a bit wary about how the loading spinners being included in the collections now means we can't lean on collection.size anymore